packages/components/src/List/ListComposition/Manager/ListManager.component.js

```diff
@@ -21,7 +23,14 @@ function Manager({
 
 	const [displayMode, setDisplayMode] = useState(initialDisplayMode || displayModesOptions[0]);
 	const [columns, setColumns] = useState([]);
-	const [visibleColumns, setVisibleColumns] = useState(initialVisibleColumns);
+
+	const { visibleColumns, setVisibleColumns, updateColumns } = useColumnsVisibility(
+		columnsVisibilityStorageKey,
+		initialVisibleColumns,
+	);
+
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	useEffect(() => updateColumns(columns), [columns]);
```

Comment:
If that's related to the hook's behavior, this useEffect should be integrated into `useColumnsVisibility()`.
It would require to provide `columns` prop but it would allow to simplify some of the code, having `columns` directly available in the custom hook (`setVisibleColumns`, `updateColumns`)